box-share-sdk/src/main/java/com/box/androidsdk/share/activities/BoxInviteCollaboratorsActivity.java

```diff
@@ -93,96 +31,17 @@ protected void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.activity_invite_collaborators);
         initToolbar();
 
-        mRoleButton = (Button) findViewById(R.id.invite_collaborator_role);
-        mRoleButton.setOnClickListener(this);
-        mAutoComplete = (MultiAutoCompleteTextView) findViewById(R.id.invite_collaborator_autocomplete);
-        mAutoComplete.setTokenizer(new MultiAutoCompleteTextView.CommaTokenizer());
-        mAdapter = new InviteeAdapter(this);
-        mAutoComplete.setAdapter(mAdapter);
-
-        Intent intent = getIntent();
-        String userId = intent.getStringExtra(EXTRA_USER_ID);
-        mSession = new BoxSession(this, userId);
-        mFolder = (BoxFolder) mShareItem;
-
-        // Get serialized roles or fetch them if they are not available
-        if (mFolder != null && mFolder.getAllowedInviteeRoles() != null) {
-            mRoles = mFolder.getAllowedInviteeRoles().toArray(new BoxCollaboration.Role[mFolder.getAllowedInviteeRoles().size()]);
-            BoxCollaboration.Role selectedRole = (BoxCollaboration.Role) intent.getSerializableExtra(EXTRA_SELECTED_ROLE);
-            setSelectedRole(selectedRole);
-        } else {
-            fetchRoles();
+        ShareController controller = new BoxShareController(new BoxApiFile(mSession),
```

Comment:
This looks like it is in both activities. Maybe we can move the logic to a parent class into its own method. That way it would be easy to override.